### ISSUE TITLE
Truncate long inlay hints

### ISSUE THREAD

**Issue by mjbvz** [microsoft/vscode#199610]
## Problem
See https://github.com/microsoft/TypeScript/issues/56541

TypeScript's inlay hints can be rather long. Previously we truncate these inlay hints at 30 characters. However we disabled this truncation when adding support for interactive inlay hints

While discussing this problem with the TS team, we realized that it would likely make more sense for VS Code to control the truncation instead of leaving it up to individual languages/extensions. That way the user experience is more consistent and an extension can't break us by sending extremely long inlay hints

The Rust extension also currently has logic to truncate their inlay hints

## Proposal
Add a default length limit to inlay hints. We could also add a setting that let's users control this limit

In addition, perhaps we can provide a way to view the full hint if you really want to. Either showing it in on hover or in a tooltip

**Comment by iambpn** [comment]
with the new TS interactive change this is much needed feature. I m working with mongoose and when enabling inlay hints, it pushes reamining code in same line to next line.

Here is one example of it:

![image](https://github.com/microsoft/vscode/assets/21078512/a8cb0eae-899c-4f21-b740-5237c716e031)

I think another possible way to solve this is to, truncate inlay hints to 30 characters by default and when `ctrl + hover` on inlay hints it expands to show full inlay hints and can interact with it. Also by default `ctrl + click` is the default way to go to definition.

**Comment by AlbertMarashi** [comment]
![image](https://github.com/microsoft/vscode/assets/17972275/7e38e62e-665a-4d02-8a89-7ad920ac7ed6)
Yerr this is pretty annoying

**Comment by hi-reeve** [comment]
yes please, with the new interactive inlay hints, this is very annoying because the code is very unreadable now

**Comment by bmacher** [comment]
Is there any progress on this as the PR (#201190) does not seem to be released yet. Code is sometimes unreadable.

**Comment by Cahmoraes** [comment]
I have the same problem, using inlay hints:
![image](https://github.com/microsoft/vscode/assets/17452938/b93f2f0a-1c73-46bd-9514-42ca1ac3e9c7)

**Comment by miszak** [comment]
> #201190 adds a first version for this. This is a bit ham-fisted and limits inlays to 43 character per line. This can be further improved by
> 
> * make the limit dynamic to the view port width
> * taking the line contents into account (e.g have a dynamic limit and allow to fill up the view port horizontally)
> * honour the different label parts and do truncation within them

I am surprised by the way the introduction of the hard limit for truncating inlay hints took place. While I perfectly understand that without the limit there are problems in the e.g. TypeScript code, it makes functionality of my language extension severely limited. 

Even when the lines are very short (but at the same time relatively rich in the inlay hint information), the inlay hint info is (unnecessarily) truncated, e.g.:
![image](https://github.com/microsoft/vscode/assets/1482027/e8e88d36-2016-432b-8c15-020479a3352c).

Would it be possible to **make the limit globally or language extension configurable in the next release(s) of Visual Studio Code**?

**Comment by felixhirt** [comment]
It would be nice if this was configurable, and if we could set a max length for individual inlay hints. Otherwise, if one of them is too long, all following ones in the same expression are hidden as well
![image](https://github.com/microsoft/vscode/assets/34443877/61dff4d7-e353-4bf2-9b32-c3909f92ea6f)

**Comment by xidianzhangjun** [comment]
> Please make the truncation configurable. I really really need inlay hints without limit.

+1.

**Comment by yangdanny97** [comment]
Seems like inlay hint truncation is implemented + configurable, and the issue can be closed?

https://code.visualstudio.com/updates/v1_94#_inlay-hint-improvements
